### PR TITLE
Add helper to generate random EAN13 strings

### PR DESCRIPTION
**Pull request by paololim**

### CODE REVIEW

File: app/io/flow/play/util/EanHelper.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  private[this] val codeLength = 12
+  private[this] val fullLength = 13
+  private[this] val random = new Random()
+
+  def generate(): String = {
+    val s = random.numeric(codeLength)
+    Seq("%s", "%s").mkString("").format(s, checksum(s))
+  }
+
+  def valid(s: String): Boolean = {
```
Comment (mbryzek) [inline app/io/flow/play/util/EanHelper.scala]: there is a chance implementation will get an error if for example you pass in "" (have not actually verified, but I think the assertion will fail)

An alternative is to call this:

def validate(s: String): Seq[String]

and return a list of error messages - one of which can be invalid checksum.

**Comment by mbryzek** [comment]
code looks great - one small comment on validate

Also- do you think EanHelper belongs in lib-play vs. just in fulfillment or tracking directly? Do we have multiple users of it?

**Comment by paololim** [comment]
@mbryzek Updated the validate function to return a list of string errors. Agreed this is probably a better approach, and one which would bubble up the reason(s) for an invalid code rather than just true vs false.

Re: services using this - right now only tracking. But I possibly will use it too in fulfillment, and if we are going to do any barcoding/upc generation in the future (catalog?), this will likely be used too.

**Comment by mbryzek** [comment]
EanHelper -> Maybe EanGenerator is better?

**Comment by paololim** [comment]
Sure, that works. And I'll put it in tracking first.

**Pull request closed without merging**